### PR TITLE
Prevent data corruption when a digitizer does not send some data

### LINKED ISSUE #30: stream-to-file: Ensure temporary loss in digitizer data does not break files

### PR DESCRIPTION
**Pull request by DanNixon**
Code review:
 - See that the `multiple_digitizers_missing_data` test accurately tests the failure case.
 - See that tests were not modified in the test cleanup commit.

Fixes #30.

### CODE REVIEW

File: stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    push_frame(
+        &mut file,
+        num_time_points,
+        0,
+        GpsTime::new(22, 205, 10, 55, 30, 0, 0, 0),
+        0,
+        1,
+    );
+
+    // push_frame(
```
Comment (Pasarus) [inline stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs:54]: Could we delete (or implement) these commented out tests?

File: stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    push_frame(
+        &mut file,
+        num_time_points,
+        2,
+        GpsTime::new(22, 205, 10, 55, 30, 40, 0, 0),
+        0,
+        1,
+    );
+
+    // push_frame(
```
Comment (Pasarus) [inline stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs:72]: Could we delete (or implement) these commented out tests?

**Review by Pasarus — CHANGES_REQUESTED** [review]
Just a question about some commented out code

File: stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs (hunk 1 above)
Comment (DanNixon) [reply stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs:54]: I'm inclined to leave them as they are, the test data in this test is a copy of the normal case multiple detectors test but with some missing data, this is the missing data.

Comment (Pasarus) [reply stream-to-file/src/file/trace/tests/multiple_digitizers_missing_data.rs:54]: Could you comment as to why this is a comment and not actually used in the test?

**Review by Pasarus — APPROVED** [review]

**Pull request merged**